test/module/irohad/ametsuchi/rocksdb_common_test.cpp

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <gtest/gtest.h>
+#include <boost/filesystem.hpp>
+
+#include "ametsuchi/impl/rocksdb_common.hpp"
+
+namespace fs = boost::filesystem;
+using namespace iroha::ametsuchi;
+
+class RocksDBTest : public ::testing::Test {
+ public:
+  void SetUp() override {
+    db_name_ = (fs::temp_directory_path() / fs::unique_path()).string();
+    db_port_ = std::make_shared<RocksDBPort>();
+    db_port_->initialize(db_name_);
+    tx_context_ = std::make_shared<RocksDBContext>();
+    db_port_->prepareTransaction(*tx_context_);
+
+    insertDb(key1_, value1_);
+    insertDb(key2_, value2_);
+    insertDb(key3_, value3_);
+    insertDb(key4_, value4_);
+    insertDb(key5_, value5_);
+  }
+
+  void TearDown() override {
+    tx_context_.reset();
+    db_port_.reset();
+    fs::remove_all(db_name_);
+  }
+
+  void insertDb(std::string_view key, std::string_view value) {
+    RocksDbCommon common(tx_context_);
+    common.valueBuffer() = value;
+    common.put(key);
+  }
+
+  std::string_view readDb(std::string_view key) {
+    RocksDbCommon common(tx_context_);
+    common.get(key);
+    return common.valueBuffer();
+  }
+
+  std::string db_name_;
+  std::shared_ptr<RocksDBPort> db_port_;
+  std::shared_ptr<RocksDBContext> tx_context_;
+
+  std::string const key1_ = "keY";
+  std::string const key2_ = "keYY";
+  std::string const key3_ = "ke1Y";
+  std::string const key4_ = "keyY";
+  std::string const key5_ = "ke";
+
+  std::string const value1_ = "vaLUe";
+  std::string const value2_ = "vaLUe2";
+  std::string const value3_ = "vaLUe3";
+  std::string const value4_ = "vaLUe4";
+  std::string const value5_ = "vaLUe5";
+};
+
+TEST_F(RocksDBTest, SimpleOperation) {
+  ASSERT_TRUE(readDb(key1_) == value1_);
+  ASSERT_TRUE(readDb(key2_) == value2_);
+  ASSERT_TRUE(readDb(key3_) == value3_);
+  ASSERT_TRUE(readDb(key4_) == value4_);
+  ASSERT_TRUE(readDb(key5_) == value5_);
+}
+
+TEST_F(RocksDBTest, SimpleDelete) {
+  RocksDbCommon common(tx_context_);
+  ASSERT_TRUE(common.del(key3_).ok());
+
```

Comment:
Try delete once more. and assert_false(ok())

Reply:
No, rocksdb just changes status of the entry to deleted. This status change will execute correctly independently of key existence.
https://github.com/facebook/rocksdb/issues/4975